Toxygates/src/panomicon/NetworkHandling.scala

```diff
@@ -36,15 +37,38 @@ class NetworkHandling(context: Context, matrixHandling: MatrixHandling) {
     netController.makeNetwork
   }
 
-  def interactionsToJson(ints: Iterable[Interaction]): Seq[Value] = {
-    ints.toSeq.map(i => {
-      writeJs(Map(
+  def networkToJson(n: Network): Map[String, Value] =
+    Map(
+      "nodes" -> nodesToJson(n.nodes().asScala),
+      "interactions" -> interactionsToJson(n.interactions().asScala)
+    )
+
+  def interactionsToJson(ints: Seq[Interaction]): Seq[Value] = {
+    ints.map(i => writeJs(Map(
         "from" -> writeJs(i.from().id()),
         "to" -> writeJs(i.to().id()),
         "label" -> writeJs(i.label()),
         "weight" -> writeJs(i.weight().doubleValue())
       ))
-    })
+    )
+  }
+
+  private def normalizeNodeType(t: String) = t match {
+    case "miRNA" => "microRNA"
```

Comment:
This is to mimic the behavior we had in the GWT application. I think this was originally done for the sake of legibility for the end user (to help distinguish from mRNA)